plugins/store/src/store.rs

```diff
@@ -290,14 +327,152 @@ impl<R: Runtime> Store<R> {
     pub fn is_empty(&self) -> bool {
         self.cache.is_empty()
     }
+
+    fn emit_change_event(&self, key: &str, value: &JsonValue) -> crate::Result<()> {
+        self.app.emit(
+            "store://change",
+            ChangePayload {
+                path: &self.path,
+                key,
+                value,
+            },
+        )?;
+        Ok(())
+    }
 }
 
-impl<R: Runtime> std::fmt::Debug for Store<R> {
+impl<R: Runtime> std::fmt::Debug for StoreInner<R> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("Store")
             .field("path", &self.path)
-            .field("defaults", &self.defaults)
             .field("cache", &self.cache)
             .finish()
     }
 }
+
+pub struct Store<R: Runtime> {
+    defaults: Option<HashMap<String, JsonValue>>,
+    serialize: SerializeFn,
+    deserialize: DeserializeFn,
+    auto_save: Option<Duration>,
+    auto_save_debounce_sender: Arc<Mutex<Option<UnboundedSender<AutoSaveMessage>>>>,
+    store: Arc<Mutex<StoreInner<R>>>,
```

Comment:
why do we need to use `Arc<Mutex>` here?

Reply:
Needed for auto save to mutate it in another thread